### ISSUE TITLE
Double changelog when root is package also

### ISSUE THREAD

**Issue by Igmat** [lerna/lerna#1215]
## Expected Behavior
I use lerna in [this repo](https://github.com/Igmat/baset) with conventional commits.
My root project is package too (`"packages": ["./", ...]`).

So when I run `lerna publish` everything is fine and I expect `CHANGELOG.md` to be updated accordingly to commits history.

## Current Behavior
But unfortunately every new version description appends twice in [this file](https://github.com/Igmat/baset/blob/9afb6423a05479f5f1bedddbf1657b6ea91c9fcd/CHANGELOG.md).

## Possible Solution
I'm not sure, but probably it could be solved by checking if root project is listed in packages, before applying changes to `CHANGELOG.md`.

## Steps to Reproduce (for bugs)
Just clone [this repo](https://github.com/Igmat/baset), run `npm install` and `lerna publish`.

<details><summary>lerna.json</summary><p>

```json
{
  "lerna": "2.7.1",
  "packages": [
    "./",
    "./packages/*",
    "./tests/*"
  ],
  "version": "0.2.1",
  "commands": {
    "publish": {
      "allowBranch": "master",
      "conventionalCommits": true,
      "ignore": [
        "*.md"
      ]
    },
    "run": {
      "ignore": "baset"
    }
  }
}
```
You may ignore `"./tests/*` - all packages there are `private` and created for integration testing only.
</p></details>

## Context
Of course, I can update `CHANGELOG.md` manually but it seems like redundant work. Also, I may misunderstood something about correct usage of lerna, if so I'll be glad if you'll point my mistakes.

## Your Environment

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 2.7.1|
| `npm --version`  | 5.3.0 |
| `yarn --version` | -- |
| `node --version` | 8.4.0 |

| OS | Version |
| --- | --- |
| Windows 10 Enterprise | 1607 |

**Comment by evocateur** [comment]
Whoops, sorry about that.

_Technically_, yes, you are allowed to have a lerna-managed package in the root (aka, right next to `lerna.json`, wherever that file happens to be). _In practice_, I discourage that because the root was originally intended (and works best as) the place to co-locate _all_ of your build/test tasks and devDependencies (and if you `--hoist`, dependencies) of all lerna-managed packages.

TL;DR: You have a `packages` directory, that's where all the packages should live (in my opinion, and according to the [principle of least astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment)).

**Comment by evocateur** [comment]
Yes, it was a bad design decision that forced many complicated internal compromises, not to mention confusion from users.

**Comment by evocateur** [comment]
This was fixed in #1217, published in v2.8.0

**Comment by nemoDreamer** [comment]
@evocateur I'm getting double CHANGELOG output in _all_ my `CHANGELOG.md` files.
I have a root `package.json` file that's private and un-named, for scripts, etc...

But:
```diff
diff --git a/packages/component-svg-icon/CHANGELOG.md b/packages/component-svg-icon/CHANGELOG.md
index 4f7919e..43b5e9f 100644
--- a/packages/component-svg-icon/CHANGELOG.md
+++ b/packages/component-svg-icon/CHANGELOG.md
@@ -3,6 +3,28 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+<a name="1.0.0"></a>
+# 1.0.0 (2018-03-22)
+
+
+
+<a name="0.2.1"></a>
+## 0.2.1 (2018-03-21)
+
+
+
+<a name="0.2.0"></a>
+# 0.2.0 (2018-03-15)
+
+
+### Features
+
+* **component-svg-icon:** add SvgIcon component ([334cd71](http://git.xxx.xxx/xxx/xxx/commits/334cd71)), closes [#60](http://git.xxx.xxx/xxx/xxx/issues/60)
+
+
+
+
+
 <a name="0.2.1"></a>
 ## 0.2.1 (2018-03-21)
 
```

---

`package.json`:
```json
{
  "license": "MIT",
  "private": true,
  "scripts": {
    "lint:js": "eslint --ext=.js,.jsx",
    "lint:json": "prettier -l",
    "lint:md": "remark --frail",
    "lint:js:all": "yarn lint:js .",
    "lint:json:all": "yarn lint:json '**/*.json'",
    "lint:md:all": "yarn lint:md .",
    "lint": "yarn lint:js:all && yarn lint:json:all && yarn lint:md:all",
    "format:js": "yarn lint:js --fix",
    "format:json": "yarn lint:json --write",
    "format:js:all": "yarn lint:js:all --fix",
    "format:json:all": "yarn format:json '**/*.json'",
    "format": "yarn format:js:all ; yarn format:json:all",
    "commit": "git-cz",
    "publish:lerna": "lerna publish",
    "publish:lerna:next": "yarn run publish:lerna --cd-version=prerelease --pre-id=next",
    "storybook": "start-storybook -p 9001 -c .storybook -s .storybook/public",
    "storybook:clean": "rm -rf build/storybook",
    "storybook:build": "yarn storybook:clean && build-storybook -s .storybook/public -o build/storybook",
    "storybook:deploy": "yarn storybook:build && storybook-to-ghpages --existing-output-dir build/storybook",
    "test": "jest -c .jest/config.js",
    "test:ci": "yarn test --no-cache --all --ci"
  },
  "devDependencies": {
    "@commitlint/cli": "^6.1.2",
    "@commitlint/config-conventional": "^6.1.2",
    "@storybook/addon-knobs": "^3.3.15",
    "@storybook/addon-options": "^3.3.15",
    "@storybook/addon-storyshots": "^3.3.15",
    "@storybook/react": "^3.3.15",
    "@storybook/storybook-deployer": "^2.3.0",
    "babel-core": "^6.26.0",
    "babel-eslint": "^8.2.2",
    "babel-jest": "^22.4.1",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "babel-plugin-transform-export-extensions": "^6.22.0",
    "babel-plugin-transform-object-rest-spread": "^6.26.0",
    "babel-preset-env": "^1.6.1",
    "babel-preset-react": "^6.24.1",
    "babel-runtime": "^6.23.0",
    "commitizen": "^2.9.6",
    "commitlint": "^6.1.2",
    "cz-lerna-changelog": "^1.2.1",
    "eslint": "^4.18.1",
    "eslint-config-airbnb": "^16.1.0",
    "eslint-config-prettier": "^2.9.0",
    "eslint-import-resolver-webpack": "^0.8.4",
    "eslint-plugin-import": "^2.9.0",
    "eslint-plugin-jsx-a11y": "^6.0.3",
    "eslint-plugin-prettier": "^2.6.0",
    "eslint-plugin-react": "^7.7.0",
    "husky": "^0.15.0-rc.8",
    "jest": "^22.4.2",
    "lerna": "^3.0.0-beta.8",
    "lint-staged": "^7.0.0",
    "object-assign": "^4.1.1",
    "prettier": "^1.11.0",
    "promise": "^8.0.1",
    "prop-types": "^15.6.1",
    "raf": "^3.4.0",
    "react": "^16.2.0",
    "react-dom": "^16.2.0",
    "remark-cli": "^5.0.0",
    "remark-preset-lint-consistent": "^2.0.1",
    "remark-preset-lint-recommended": "^3.0.1",
    "resolve": "^1.5.0",
    "whatwg-fetch": "^2.0.3"
  },
  "workspaces": [
    "examples/*",
    "packages/*"
  ],
  "config": {
    "commitizen": {
      "path": "./node_modules/cz-lerna-changelog"
    }
  },
  "commitlint": {
    "extends": [
      "@commitlint/config-conventional"
    ]
  },
  "husky": {
    "hooks": {
      "pre-commit": "lint-staged",
      "commit-msg": "commitlint --edit"
    }
  },
  "lint-staged": {
    "linters": {
      "**/*.{js,jsx}": [
        "yarn format:js",
        "git add"
      ],
      "**/*.json": [
        "yarn format:json",
        "git add"
      ],
      "**/*.md": [
        "yarn lint:md",
        "git add"
      ]
    },
    "concurrent": false
  }
}
```

`lerna.json`:
```json
{
  "lerna": "2.9.0",
  "packages": [
    "examples/*",
    "packages/*"
  ],
  "version": "1.0.0",
  "npmClient": "yarn",
  "useWorkspaces": true,
  "conventionalCommits": true,
  "forcePublish": true,
  "registry": "https://xxx.xxx.xxx/artifactory/api/npm/xxx"
}
```
